grammar fix (#2329)
arbor-sim/arbor: doc/fileformat/swc.rst

**Every sample must have the same SWC identifier (tag) as its parent, except for
samples whose parent is tagged as soma**:

> This enforces that axons, [basel? is missing] dendrites, and apical dendrites can only attach to the soma.

Reviewer: ```suggestion
This enforces that axonal and dendritic segments can only attach to the soma.
```
Yes, both apical and basal, so let's compress this to encompass both.

  ``x=x0`` and the third at ``x=x0+r``, to form a single section.
* All dendrites, axons, and apical dendrites are attached to the center of the soma with "zero resistance wires".

**The axon, dendrite, and apical sub-trees follow special rules for attachment to
the soma**: By default, the sub-tree starts at the first sample with the

> dendrite, axon, or apical tag [what is apical tag?], and not at the parent location on the soma, and

Reviewer: ```suggestion
a tag indicating a non-soma segment, and not at the parent location on the soma, and
```
It's a slightly confusing way of stating what a segment layout like this
```
(soma) - (soma) - (axon) - (axon)
```
should be, where `()` indicates a sample of `(x, y, z, r, tag)` where I have just shown
the tags. We need to convert this from a list of points to a list of segments with attached 
regions (soma, axon, ...). In particular we get this
```
(soma) [soma] (soma) [soma] (axon) [axon] (axon)
```
and not
```
(soma) [soma] (soma) [axon] (axon) [axon] (axon)
```
Both would have been possible solutions.

Reviewer: We are really getting into the intricacies of both Neuron and Arbor here...
